packages/grid/x-data-grid-generator/src/renderer/renderEditStatus.tsx

```diff
@@ -38,14 +38,18 @@ function EditStatus(props: GridRenderEditCellParams) {
   const { id, value, api, field } = props;
 
   const handleChange: SelectProps['onChange'] = (event) => {
-    api.setEditCellValue({ id, field, value: event.target.value }, event);
+    api.setEditCellValue({ id, field, value: event.target.value as any }, event);
     api.commitCellChange({ id, field });
     api.setCellMode(id, field, 'view');
 
     if ((event as any).key) {
       // TODO v6: remove once we stop ignoring events fired from portals
       const params = api.getCellParams(id, field);
-      api.publishEvent(GridEvents.cellNavigationKeyDown, params, event);
+      api.publishEvent(
+        GridEvents.cellNavigationKeyDown,
+        params,
+        event as any as React.KeyboardEvent<HTMLElement>,
```

Comment:
What would you think about using a [type predicates](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates) to avoid the `as any as`?

```
function isKeyboardEvent(event: React.KeyboardEvent<HTMLElement> | any): event is React.KeyboardEvent<HTMLElement>  {
  return (event as React.KeyboardEvent<HTMLElement> ).key !== undefined;
}
```

I do not know if it brings something or if it just for the esthetic

Reply:
I need the `as any` because the `SelectProps['onChange']` `event` params is a `changeEvent` so it technically can't be a `keyboardEvent`

We talked about it with @m4theushw a while back when I asked him if there was some scenario where is was actually a `keyboardEvent` and for him the typing from the core is imprecise for this param.

Honestly I did not dif deeper.
But as `changeEvent` do not extend `keyboardEvent` in any way, I will require the `as any` to trick TypeScript event with type predicates.

Unless I'm missing something, I you find a way to avoid that ugly  `as any` I'm very interested :)

Reply:
Don't think I can help with such a tricky typescript issue 😅